target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4019-qxwlan-e2600ac.dtsi

```diff
@@ -0,0 +1,262 @@
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+
+	model = "Qxwlan E2600AC";
+	compatible = "qxwlan,e2600ac", "qcom,ipq4019";
+
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x10000000>; /* 256MB */
+	};
+
+	soc {
+		mdio@90000 {
+			status = "okay";
+			pinctrl-0 = <&mdio_pins>;
+			pinctrl-names = "default";
+		};
+
+		ess-psgmii@98000 {
+			status = "okay";
+		};
+
+		counter@4a1000 {
+			compatible = "qcom,qca-gcnt";
+			reg = <0x4a1000 0x4>;
+		};
+
+		tcsr@1949000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1949000 0x100>;
+			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+		};
+
+		tcsr@194b000 {
+			/* select hostmode */
+			compatible = "qcom,tcsr";
+			reg = <0x194b000 0x100>;
+			qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+			status = "okay";
+		};
+
+		ess_tcsr@1953000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1953000 0x1000>;
+			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+		};
+
+		tcsr@1957000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1957000 0x100>;
+			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+		};
+
+		usb2: usb2@60f8800 {
+			status = "okay";
+		};
+
+		serial@78af000 {
+			pinctrl-0 = <&serial_0_pins>;
+			pinctrl-names = "default";
+			status = "okay";
+		};
+
+		serial@78b0000 {
+			pinctrl-0 = <&serial_1_pins>;
+			pinctrl-names = "default";
+			status = "okay";
+		};
+
+		i2c@78b7000 { /* BLSP1 QUP2 */
+			pinctrl-0 = <&i2c_0_pins>;
+			pinctrl-names = "default";
+
+			status = "okay";
+		};
+
+		usb3: usb3@8af8800 {
+			status = "okay";
+		};
+
+		crypto@8e3a000 {
+			status = "okay";
+		};
+
+		watchdog@b017000 {
+			status = "okay";
+		};
+
+		ess-switch@c000000 {
+			status = "okay";
+		};
+
+		edma@c080000 {
+			status = "okay";
+		};
+
+		aliases {
+			led-boot = &status;
+			led-failsafe = &status;
+			led-upgrade = &status;
+		};
+
+		gpio-leds {
+			compatible = "gpio-leds";
+
+			led1 {
+				label = "e2600ac:green:ds20";
+				gpios = <&tlmm 50 GPIO_ACTIVE_LOW>;
+				default-state = "off";
+			};
+			led2 {
+				label = "e2600ac:green:ds14";
+				gpios = <&tlmm 36 GPIO_ACTIVE_LOW>;
+				default-state = "off";
+			};
+			status:led3 {
+				label = "e2600ac:green:ds17";
+				gpios = <&tlmm 32 GPIO_ACTIVE_HIGH>;
+				default-state = "off";
+			};
+			led4 {
+				label = "e2600ac:green:ds19";
+				gpios = <&tlmm 51 GPIO_ACTIVE_LOW>;
+				default-state = "off";
+			};
+			led5 {
+				label = "e2600ac:green:ds18";
+				gpios = <&tlmm 30 GPIO_ACTIVE_LOW>;
+				default-state = "off";
+			};
+			led6 {
+				label = "e2600ac:green:ds21";
+				gpios = <&tlmm 31 GPIO_ACTIVE_LOW>;
+				default-state = "off";
+			};
+		};
+
+		gpio_keys {
```

Comment:
If you want, you could rename the node to "keys" to follow the "generic
names recommendation" of the Device Tree V0.2 Spec Section 
"2.2.2 Generic Names Recommendation."